R/zchunk_LB1322.Fert.R

```diff
@@ -45,88 +57,386 @@ module_energy_LB1322.Fert_DISABLED <- function(command, ...) {
+    L1322.Fert_ALL_MtN_R_F_Y_adj %>%
+      select(GCAM_region_ID, sector, fuel, year, value = Fert_Prod_MtN_adj) ->
+      L1322.Fert_Prod_MtN_R_F_Y # Note that this is a final output table
+
+    # Creating final output table "Fertilizer input-output coefs by GCAM region / fuel (base year techs only) / year"
+    L1322.Fert_ALL_MtN_R_F_Y_adj %>%
+      select(GCAM_region_ID, sector, fuel, year, value = intensity_GJkgN) ->
+      L1322.IO_R_Fert_F_Yh # Note that this is a final output table
+
+    # Creating final output table "Adjusted industrial feedstock use by GCAM region / fuel / year"
+    # We already calcuated the values. We only need to add a column for the sector and strip out unnecessary columns.
+    L1322.Fert_ALL_MtN_R_F_Y_adj %>%
+      mutate(sector = "industry_feedstocks") %>%
+      select(GCAM_region_ID, sector, fuel, year, value = in_indfeed_netFert) ->
+      L1322.in_EJ_R_indfeed_F_Yh # Note that this is a final output table
+
+    # Extract the values we calculated for industrial energy use. These values will be used for its respective final table.
+    L1322.Fert_ALL_MtN_R_F_Y_adj %>%
+      select(GCAM_region_ID, fuel, year, value = in_indenergy_netFert) ->
+      L1322.in_EJ_R_indenergy_Ffert_Yh
+
+    # Creating final output table "Adjusted industrial energy use by GCAM region / fuel / year"
+    # We are starting with the input industrial energy table and replacing a portion of the values we calcuated in this chunk.
+    L1321.in_EJ_R_indenergy_F_Yh %>%
+      rename(value_indenergy = value) %>%
+      # left_join_error_no_match cannot be used because the joining tibble has less rows, so NAs will be introduced.
+           # This will be helpful as a filter in the following step.
+      left_join(L1322.in_EJ_R_indenergy_Ffert_Yh, by = c("GCAM_region_ID", "fuel", "year")) %>% # Join values we calculated
+      # Replace any NAs with original industrial energy value
+      mutate(value = replace(value, is.na(value), value_indenergy[is.na(value)])) %>%
+      select(GCAM_region_ID, sector, fuel, year, value) ->
+      L1322.in_EJ_R_indenergy_F_Yh # Note that this is a final output table.
+
+    # -----------------------------------------------------------------------------------------------------------------
+
+    # Calculate fertilizer non-energy costs by technology
+         # These technologies include gas, gas with CCS, coal, coal with CCS, and oil
+
+    # First, calculate gas cost per kg N
+         # Calculating non-energy costs for gas technology as USA market fertilizer price minus GCAM fuel costs
+         # Calculate the gas price as the sum of resource costs plus intermediate sectoral mark-ups
+         # 2005 is used as the fertilizer base price. We will interpolate for 2005 in the latter cost tables to make
+         # sure this year is included
+    A10.rsrc_info %>%
+      filter(resource == "natural gas") %>%
+      summarise(value = sum(`2005`)) %>% # Ensuring no duplicates
```

Comment:
I had to add a couple lines, but I think it worked well and maintains consistency. Thanks.